runner/src/pipeline.rs

```diff
@@ -199,35 +216,36 @@ impl Layer {
             LayerFunctionDef::Some(func) => func.call(input, original_input, context).await?,
             LayerFunctionDef::UseDefault => {
                 if let Some(default) = default {
-                    default.context.clone()
+                    Python::with_gil(|py| default.context.clone_ref(py))
```

Comment:
I mean basically this whole function happens with references to the gil, so maybe it should just be wrapped in Python::with_gil, but I think that's outside of the scope of this PR. This should just be about migrating